packages/devtools_app/lib/src/screens/memory/memory_heap_treemap.dart

```diff
@@ -4,57 +4,49 @@
 
 import 'package:flutter/material.dart' hide TextStyle;
 import 'package:flutter/widgets.dart' hide TextStyle;
-import 'package:provider/provider.dart';
 
 import '../../charts/treemap.dart';
 import '../../primitives/auto_dispose_mixin.dart';
 import '../../shared/common_widgets.dart';
 import '../../shared/theme.dart';
+import '../../shared/utils.dart';
 import 'memory_controller.dart';
 import 'primitives/predefined_classes.dart';
 
 class MemoryHeapTreemap extends StatefulWidget {
-  const MemoryHeapTreemap(this.controller);
-
-  final MemoryController controller;
+  const MemoryHeapTreemap();
 
   @override
-  MemoryHeapTreemapState createState() => MemoryHeapTreemapState(controller);
+  MemoryHeapTreemapState createState() => MemoryHeapTreemapState();
 }
 
 class MemoryHeapTreemapState extends State<MemoryHeapTreemap>
-    with AutoDisposeMixin {
-  MemoryHeapTreemapState(this._controller);
-
+    with
+        AutoDisposeMixin,
+        ProvidedControllerMixin<MemoryController, MemoryHeapTreemap> {
   InstructionsSize? _sizes;
 
-  bool _initialized = false;
-  late MemoryController _controller;
-
   TreemapNode? root;
 
   @override
   void didChangeDependencies() {
     super.didChangeDependencies();
 
     // TODO(terry): Unable to short-circuit need to investigate why?
```

Comment:
Is this TODO needed anymore?

Reply:
I am not sure. Leaving the TODO since this is out of scope of this PR.